crates/solvers/src/domain/solver/dex/mod.rs

```diff
@@ -33,10 +40,17 @@ pub struct Dex {
 
     /// Parameters used to calculate the revert risk of a solution.
     risk: domain::Risk,
+
+    /// Handles 429 Too Many Requests error with a retry mechanism
+    rate_limiter: Arc<RateLimiter>,
 }
 
 impl Dex {
-    pub fn new(dex: infra::dex::Dex, config: infra::config::dex::Config) -> Self {
+    pub fn new(
+        dex: infra::dex::Dex,
+        config: infra::config::dex::Config,
```

Comment:
Should the rate limiter config be part of `infra::config::dex::Config` and then be created in this constructor instead of passed in from above? Otherwise we may specify it for solvers that don't even make use of a rate limiter, which may be confusing.

Reply:
Migrated to the `infra::config::dex::Config`